src/filter.cpp

```diff
@@ -677,30 +677,34 @@ Option<bool> toFilter(const std::string& expression,
         NUM_COMPARATOR str_comparator = CONTAINS;
         auto apply_not_equals = false;
         if (raw_value[0] == '=') {
-            // string filter should be evaluated in strict "equals" mode
             str_comparator = EQUALS;
             while (++filter_value_index < raw_value.size() && raw_value[filter_value_index] == ' ');
         } else if (raw_value.size() >= 2 && raw_value[0] == '!') {
             if (raw_value[1] == '=') {
                 str_comparator = NOT_EQUALS;
                 filter_value_index++;
             }
-
             apply_not_equals = true;
             while (++filter_value_index < raw_value.size() && raw_value[filter_value_index] == ' ');
         }
         if (filter_value_index == raw_value.size()) {
-            return Option<bool>(400, "Error with filter field `" + _field.name +
-                                     "`: Filter value cannot be empty.");
+            return Option<bool>(400, "Error with filter field `" + _field.name + "`: Filter value cannot be empty.");
         }
-        if (raw_value[filter_value_index] == '[' && raw_value[raw_value.size() - 1] == ']') {
+
+        std::string value_part = raw_value.substr(filter_value_index);
+
+        if (value_part.length() > 1 && value_part.front() == '"' && value_part.back() == '"') {
+            str_comparator = CONTAINS_PHRASE;
+            value_part = value_part.substr(1, value_part.length() - 2);
+            StringUtils::replace_all(value_part, "\\\"", "\"");
```

Comment:
I don't follow this statement. We have already removed the double quotes enclosing the filter value in the previous line.